builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -293,25 +292,10 @@ func validateS3BucketObjectAclType(v interface{}, k string) (ws []string, errors
 		s3.ObjectCannedACLBucketOwnerFullControl: true,
 	}
 
-	sentenceJoin := func(m map[string]bool) string {
-		keys := make([]string, 0, len(m))
-		for k := range m {
-			keys = append(keys, fmt.Sprintf("%q", k))
-		}
-		sort.Strings(keys)
-
-		length := len(keys)
-		words := make([]string, length)
-		copy(words, keys)
-
-		words[length-1] = fmt.Sprintf("or %s", words[length-1])
-		return strings.Join(words, ", ")
-	}
-
 	if _, ok := cannedAcls[value]; !ok {
 		errors = append(errors, fmt.Errorf(
 			"%q contains an invalid canned ACL type %q. Valid types are either %s",
-			k, value, sentenceJoin(cannedAcls)))
+			k, value, sentenceJoin("or", cannedAcls)))
```

Comment:
I'm tempted to say we shouldn't touch this as part of this PR. It isn't related at all to the API Gateway validator. I think this should be a separate PR IMO